crates/runtime/tests/databricks_delta/mod.rs

```diff
@@ -0,0 +1,129 @@
+
+fn make_dataset(path: &str, name: &str) -> Dataset {
+    let mut dataset = Dataset::new(format!("databricks:{path}"), name.to_string());
+    dataset.params = Some(get_params());
+    dataset
+}
+
+fn get_params() -> Params {
+    Params::from_string_map(
+        vec![
+            (
+                "databricks_endpoint".to_string(),
+                "${ env:DATABRICKS_HOST }".to_string(),
+            ),
+            (
+                "databricks_token".to_string(),
+                "${ env:DATABRICKS_TOKEN }".to_string(),
+            ),
+            (
+                "databricks_aws_secret_access_key".to_string(),
+                "${ env:AWS_DATABRICKS_DELTA_SECRET_ACCESS_KEY }".to_string(),
+            ),
+            (
+                "databricks_aws_access_key_id".to_string(),
+                "${ env:AWS_DATABRICKS_DELTA_ACCESS_KEY_ID }".to_string(),
+            ),
+            ("client_timeout".to_string(), "120s".to_string()),
+            ("mode".to_string(), "delta_lake".to_string()),
+        ]
+        .into_iter()
+        .collect(),
+    )
+}
+
+#[tokio::test]
+async fn databricks_delta_lake_integration_test() -> Result<(), anyhow::Error> {
+    type QueryTests<'a> = Vec<(&'a str, &'a str, Option<Box<ValidateFn>>)>;
+    let _tracing = init_tracing(Some("integration=debug,info"));
+
+    test_request_context()
+        .scope(async {
+            let app = AppBuilder::new("databricks_delta_lake_connector")
+                .with_dataset(make_dataset(
+                    "spiceai_sandbox.integration.delta_all_types_table",
+                    "datatypes",
+                ))
+                .build();
+
+            let status = status::RuntimeStatus::new();
+            let df = get_test_datafusion(Arc::clone(&status));
+
+            let mut rt = Runtime::builder()
+                .with_datafusion(df)
+                .with_app(app)
+                .build()
+                .await;
+
+            // Set a timeout for the test
+            tokio::select! {
+                () = tokio::time::sleep(std::time::Duration::from_secs(10)) => {
+                    return Err(anyhow::anyhow!("Timed out waiting for datasets to load"));
+                }
+                () = rt.load_components() => {}
+            }
+
+            let queries: QueryTests = vec![(
+                "SELECT * FROM datatypes",
+                "select",
+                Some(Box::new(|result_batches| {
+                    for batch in &result_batches {
+                        assert_eq!(batch.num_columns(), 15, "num_cols: {}", batch.num_columns());
+                        assert_eq!(batch.num_rows(), 1, "num_rows: {}", batch.num_rows());
+                    }
+
+                    // snapshot the values of the results
+                    let results = arrow::util::pretty::pretty_format_batches(&result_batches)
+                        .expect("should pretty print result batch");
+                    insta::with_settings!({
```

Comment:
Please review if `insta::with_settings!` is  required vs we can avoid this for simplicity

Reply:
It's needed for snapshots to be placed in the right directory